### PR TITLE
github: cache all npm installs to improve CI performance

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

**Review by jasonLaster — APPROVED** [review]

File: .github/actions/test/action.yml [hunk 1]
```diff
@@ -3,8 +3,7 @@ description: "Run end-to-end tests. Note that this action assumes that Node and
 runs:
   using: "composite"
   steps:
-    - name: Install dependencies
-      run: npm install
+    - uses: bahmutov/npm-install@v1
```
Comment (jcmorrow) [inline .github/actions/test/action.yml:6]: What do you think about using something a little more official like the GitHub action for node?  https://github.com/actions/setup-node#caching-global-packages-data

**Comment by jasonLaster** [comment]
NOTE the official version does not cache node_modules

**Comment by jcmorrow** [comment]
> NOTE the official version does not cache node_modules

Huh... That seems odd? Oh well, I guess we go with this then 👍

**Comment by hbenl** [comment]
We could run into the issue that `npm install` fails sporadically without `--force` again, but I think it's worth a try.

**Pull request merged**